src/sentry/integrations/slack/message_builder/metric_alerts.py

```diff
@@ -0,0 +1,76 @@
+from datetime import datetime
+from typing import Optional
+
+from sentry.incidents.models import AlertRule, Incident, IncidentStatus
+from sentry.integrations.metric_alerts import metric_alert_attachment_info
+from sentry.integrations.slack.message_builder import INCIDENT_COLOR_MAPPING, SlackBody
+from sentry.integrations.slack.message_builder.base.base import SlackMessageBuilder
+from sentry.utils.dates import to_timestamp
+
+
+def get_footer(
+    incident_triggered_date: Optional[datetime], last_triggered_date: Optional[datetime]
+) -> str:
+    if incident_triggered_date:
+        return "<!date^{:.0f}^Sentry Incident - Started {} at {} | Sentry Incident>".format(
+            to_timestamp(incident_triggered_date), "{date_pretty}", "{time}"
+        )
+
+    if last_triggered_date:
+        return "<!date^{:.0f}^Metric Alert - Last Triggered {} at {} | Metric Alert>".format(
+            to_timestamp(last_triggered_date), "{date_pretty}", "{time}"
+        )
+
+    return "Metric Alert"
+
+
+class SlackMetricAlertMessageBuilder(SlackMessageBuilder):
+    def __init__(
+        self,
+        alert_rule: AlertRule,
+        incident: Optional[Incident] = None,
+        new_status: Optional[IncidentStatus] = None,
+        metric_value: Optional[int] = None,
+    ) -> None:
+        """
+        Builds a metric alert attachment for slack unfurling.
+
+        :param alert_rule: The `AlertRule` for which to build the attachment.
+        :param incident: The `Incident` for which to build the attachment.
+        :param [new_status]: The new status of a metric alert incident
+        :param [metric_value]: The value of the metric that triggered this alert to
+            fire. If not provided we'll attempt to calculate this ourselves.
+        """
+        super().__init__()
+        self.alert_rule = alert_rule
+        self.incident = incident
+        self.metric_value = metric_value
+        self.new_status = new_status
+
+    def build(self, unfurl: bool = False) -> SlackBody:
+        data = metric_alert_attachment_info(
+            self.alert_rule, self.incident, self.new_status, self.metric_value, unfurl=unfurl
+        )
+
+        return self._build(
+            actions=[],
+            color=INCIDENT_COLOR_MAPPING.get(data["status"]),
+            fallback=data["title"],
+            fields=[],
+            footer=get_footer(data["date_started"], data["last_triggered_date"]),
+            text=data["text"],
+            title=data["title"],
+            title_link=data["title_link"],
+        )
+
+
+def build_metric_alert_attachment(
+    alert_rule: AlertRule,
+    incident: Optional[Incident] = None,
+    metric_value: Optional[int] = None,
+    unfurl: bool = False,
+) -> SlackBody:
+    """@deprecated"""
```

Comment:
how is new code already deprecated

Reply:
This is just taken from the incidents file? I don't really know

Reply:
You can just replace the usage in `src/sentry/integrations/slack/unfurl/build_metric_alert_attachment.py` with:

```py
return {
    link.url: SlackMetricAlertMessageBuilder(
        alert_rule=alert_rule_map[link.args["alert_rule_id"]],
        incident=incident_map.get(link.args["incident_id"]),
    ).build(unfurl=True)
    for link in links
    if link.args["alert_rule_id"] in alert_rule_map
}
```

Reply:
@mgaeta I have a vague memory of this being deprecated but I'm not sure what it was replaced with, do you know? Is it okay for someone to keep using the `SlackMessageBuilder`?